### ISSUE TITLE
Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by joshcsullivan** [marcelklehr/floccus#547]
**Describe the bug**
I stupidly updated NC Bookmarks to 3.1.0 without checking how long it's been out (~2 hours), and floccus immediately came up with this error in Brave (Chromium): `Cannot read property 'filter' of undefined`. Firefox (which uses another server folder) comes back with: `this.children is undefined`. I've tried removing the account, removing the local bookmarks and setting sync up again, but still having the same issue.

**To Reproduce**
1. Upgrade to NC Bookmarks 3.1.0
2. Either force sync or wait for floccus to sync, and it will come back with an error.

**Expected behavior**
floccus syncs local changes to NC Bookmarks.

**Screenshots**
![image](https://user-images.githubusercontent.com/5774100/83634696-f2ffec00-a570-11ea-946c-c690e4f71869.png)

**Desktop (please complete the following information):**
 - OS: Windows Enterprise 1909
 - Browser: Brave (Chromium)
 - Browser Version:Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit)
 - Floccus version: 4.1.0
 - Floccus sync method: NextCloud Folders

**Server (please complete the following information):**
 - OS: Ubuntu 18.04.4 LTS
 - Nextcloud version: 18.0.4
 - Bookmarks app version: 3.1.0

Note: Please write down the actual version numbers instead of writing 'latest'.

**Debug log**
It is often useful to provide a debug log file along with the issue. You can obtain a debug log of the most recent sync runs by clicking 'Debug log' in the bottom right corner of the floccus panel in your browser.

Note: As your debug log usually contains large parts of your bookmark data, instead of attaching it publicly to the issue here, I recommend uploading it privately at the following URL https://klehranlage.net/index.php/s/7nn82NSBsfQKdsJ

- [X] Debug log provided
File Name: `GitHub_Issue-547_floccus-4.1.0-2020-06-03`

**Additional context**
I also use [nextBookmark](https://gitlab.com/altepizza/nextbookmark) on iOS, and it is syncing correctly, just as a point of reference that NextCloud sync is still working.

Cheers,
Josh

**Comment by crh0831** [comment]
Same.  Nothing new to contribute except that it's affecting me the exact same way.  It appears the update to Nextcloud Bookmarks 3.1.0 is the casue.

**Comment by marcelklehr** [comment]
Yep. v3.1.1 is out already.

**Comment by marcelklehr** [comment]
It's a double bug. A bug in nextcloud bookmarks caused a bug in floccus :scream_cat:

**Comment by Erebus999** [comment]
Same here

**Comment by crh0831** [comment]
Maybe I'm doing this wrong, but since I haven't receive the notification to install the **Nextcloud Bookmarks** update I looked to see if I could manually install the fix.  The diff between 3.1.0 and 3.1.1 (other than version numbers) is one character on one line. 

See https://github.com/nextcloud/bookmarks/compare/v3.1.0...v3.1.1

I made the change on line 665 (per the diff file) and restarted apache, but floccus still isn't working.

Again, maybe I did the patch wrong (I only changed line 665).  

Anyone know how to force Nextcloud to look for app updates?
